.travis.yml

```diff
@@ -1,16 +1,48 @@
+dist: bionic
 language: python
 sudo: false
 cache:
+  pip: true
   directories:
   - eggs
-  - downloads
 python:
-  - 2.7
+  - "2.7"
 matrix:
+  include:
+    # - python: "2.7"
+    #   env: PLONE_VERSION=43
+    # - python: "2.7"
+    #   env: PLONE_VERSION=51
+    # - python: "2.7"
+      env: PLONE_VERSION=52
+    - python: "3.7"
+      env: PLONE_VERSION=52
+  sudo: true
   fast_finish: true
+before_install:
+  - sudo apt-get install -y firefox-geckodriver
+  - virtualenv -p `which python` .
+  - bin/pip install -r requirements.txt -c constraints_plone$PLONE_VERSION.txt
+  - cp test_plone$PLONE_VERSION.cfg buildout.cfg
+
 install:
-  - python bootstrap.py
-  - bin/buildout -N buildout:download-cache=downloads annotate
-  - bin/buildout -N buildout:download-cache=downloads
+  - bin/buildout -N -t 3 code-analysis:return-status-codes=True annotate
+  - bin/buildout -N -t 3 code-analysis:return-status-codes=True
+
+before_script:
+- 'export DISPLAY=:99.0'
+- Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+- sleep 3
+- firefox -v
+
 script:
-  - bin/test
+# Run code-analysis, except on Python 3.6, which mysteriously fails to find zc.buildout.
+  - python --version 2> /dev/stdout | grep 3.6 || bin/code-analysis
+  - bin/test --all
+
+after_success:
+  - bin/createcoverage --output-dir=parts/test/coverage
+  - bin/pip install coverage
+  - bin/python -m coverage.pickle2json
```

Comment:
I think pickle2json is not needed anymore, and may break. But not sure.
I don't know if the current coverage setup is okay. I have seen lots of projects where it is not, where a coverage report might be created just fine, but the sending to coveralls fails, probably because the project has not been registered yet. I have not cared enough to check for those projects. Maybe it is fine here.